pkg/querier/frontend2/frontend.proto

```diff
@@ -10,36 +10,6 @@ import "github.com/weaveworks/common/httpgrpc/httpgrpc.proto";
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
-// Scheduler interface exposed to Queriers.
-service SchedulerForQuerier {
-    // After calling this method, both Querier and Scheduler enter a loop, in which querier waits for
-    // "SchedulerToQuerier" messages containing HTTP requests and processes them. After processing the request,
-    // querier signals that it is ready to accept another one by sending empty QuerierToScheduler message.
-    //
-    // Long-running loop is used to detect broken connection between scheduler and querier. This is important
-    // for scheduler to keep a list of connected queriers up-to-date.
-    rpc QuerierLoop(stream QuerierToScheduler) returns (stream SchedulerToQuerier) { };
-}
-
-// Querier reports its own clientID when it connects, so that scheduler knows how many *different* queriers are connected.
-// To signal that querier is ready to accept another request, querier sends empty message.
-message QuerierToScheduler {
-    string querierID = 1;
-}
-
-message SchedulerToQuerier {
-    // Query ID as reported by frontend. When querier sends the response back to frontend (using frontendAddress),
-    // it identifies the query by using this ID.
-    uint64 queryID = 1;
-    httpgrpc.HTTPRequest httpRequest = 2;
-
-    // Where should querier send HTTP Response to (using FrontendForQuerier interface).
-    string frontendAddress = 3;
-
-    // User who initiated the request. Needed to send reply back to frontend.
-    string userID = 4;
-}
-
 // Frontend interface exposed to Queriers. Used by queriers to report back the result of the query.
```

Comment:
I think it's little unfortunate that frontend.proto is now split into two files – I considered it a plus that single proto file defined the entire workflow.

Reply:
I personally believe every Cortex service should expose its own proto. This way, you can have a clear separation of concerns: "which service exposes what". I don't think the architecture is more clear to understand just because all RPC are defined in a single file. On the contrary, if each service exposes its own `.proto` you can have a better understanding of which service exposes a specific RPC. Overall architecture and workflow should be documented elsewhere IMO, and definitely not in a protobuf-style (better to explain it in plain english, with diagrams like you did in the query-scheduler PR description).